test.js

```diff
@@ -5,27 +5,202 @@ import tempy from 'tempy';
 import execa from 'execa';
 import fileIcon from '.';
 
+const processId = async app => {
+	const {stdout} = await execa('pgrep', [app]);
+	return parseInt(stdout.split('\n')[0], 10);
+};
+
+const tmpFile = () => tempy.file({extension: 'png'});
+
+test('argument validation', async t => {
+	const finderPid = await processId('Finder');
+
+	await t.notThrowsAsync(() => fileIcon.buffer('Safari'));
+	await t.notThrowsAsync(() => fileIcon.buffer(finderPid));
+	await t.notThrowsAsync(() => fileIcon.buffer(['Safari', 'Finder']));
+	await t.notThrowsAsync(() => fileIcon.buffer(['Safari', finderPid]));
+
+	await t.throwsAsync(() => fileIcon.buffer({}), TypeError);
+	await t.throwsAsync(() => fileIcon.buffer([{}]), TypeError);
+	await t.throwsAsync(() => fileIcon.buffer(['Safari', {}]), TypeError);
+
+	await t.notThrowsAsync(() => fileIcon.file('Safari', {destination: tmpFile()}));
+	await t.notThrowsAsync(() => fileIcon.file(finderPid, {destination: tmpFile()}));
+	await t.notThrowsAsync(() => fileIcon.file(['Safari', 'Finder'], {destination: [tmpFile(), tmpFile()]}));
+	await t.notThrowsAsync(() => fileIcon.file(['Safari', finderPid], {destination: [tmpFile(), tmpFile()]}));
+
+	await t.throwsAsync(() => fileIcon.file(['Safari']), TypeError);
+	await t.throwsAsync(() => fileIcon.file(['Safari'], {destination: tmpFile()}), TypeError);
+	await t.throwsAsync(() => fileIcon.file(['Safari', 'Finder'], {destination: [tmpFile()]}), TypeError);
+	await t.throwsAsync(() => fileIcon.file(['Safari', 'Finder'], {destination: [tmpFile(), tmpFile(), tmpFile()]}), TypeError);
+
+	await t.throwsAsync(() => fileIcon.file({}, {destination: tmpFile()}), TypeError);
+	await t.throwsAsync(() => fileIcon.file([{}], {destination: [tmpFile()]}), TypeError);
+	await t.throwsAsync(() => fileIcon.file(['Safari', {}], {destination: [tmpFile(), tmpFile()]}), TypeError);
+});
+
 test('app name', async t => {
 	t.is(fileType(await fileIcon.buffer('Safari')).ext, 'png');
 });
 
+test('array of single app name', async t => {
+	const apps = ['Finder'];
+	const buffers = await fileIcon.buffer(apps);
+
+	t.is(fileType(buffers[0]).ext, 'png');
```

Comment:
Can you also check the length of the array in all these tests too? To ensure the count is correct.